File: Quiz.Web/src/stores/responseDtos/extendedQuizResponseDto.ts
import type { extendedQuestionResponseDto } from '@stores/responseDtos/extendedQuestionResponseDto'

export interface extendedQuizResponseDto {
    id: number,
    pin: string,
    title: string,
    questions: extendedQuestionResponseDto[],
    participants: string[],
    isStarted: boolean,
    indexOfActiveQuestion: number,
    isPublished: boolean
}

File: Quiz.Shared/Responses/ExtendedQuizResponseDto.cs
namespace Quiz.Shared.Responses
{
    public record ExtendedQuizResponseDto
    {
        public required int Id { get; init; }
        public string? Pin { get; init; }
        public required string Title { get; init; }
        public required ICollection<ExtendedQuestionResponseDto> Questions { get; init; }
        public required List<string> Participants { get; init; }
        public required bool IsStarted { get; init; }
        public required int IndexOfActiveQuestion { get; init; }
        public required bool IsPublished { get; init; }
    }
}
